pennylane/devices/preprocess.py

```diff
@@ -479,7 +479,9 @@ def sample_measurements(m):
         and not isinstance(meas := op.hyperparameters["measurement"], qml.measurements.StateMP)
     ]
 
-    if tape.shots:
+    shots = qml.measurements.Shots(tape.shots)
+
+    if shots.total_shots is not None:
```

Comment:
I'm curious about this change. `tape.shots` should always be of type `Shots`

Reply:
Not with the legacy device iirc